### PR TITLE
JDK-8328398 : Convert java/awt/im/4490692/bug4490692.html applet test to main

### PR DESCRIPTION
**Pull request by honkar-jdk**
Following test is converted from applet to main using PassFailJFrame.

The original test had ignore tag - **The fix for this problem was backed out for 5057184.  Remove this "ignore" tag when it is fixed again.**
I believe this was fixed and the test works correctly on Linux platforms, hence ignore tag has been removed.

This is a semi-automated test, the first check is manual and the 2nd one is automated. Details in the test instructions.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328398](https://bugs.openjdk.org/browse/JDK-8328398): Convert java/awt/im/4490692/bug4490692.html applet test to main (**Bug** - P4)

### Reviewers
 * [Alisen Chung](https://openjdk.org/census#achung) (@alisenchung - Committer)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**) ⚠️ Review applies to [03cc8e41](https://git.openjdk.org/jdk/pull/18402/files/03cc8e41212ac497d41885054f4c6a17677cb2e1)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18402/head:pull/18402` \
`$ git checkout pull/18402`

Update a local copy of the PR: \
`$ git checkout pull/18402` \
`$ git pull https://git.openjdk.org/jdk.git pull/18402/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18402`

View PR using the GUI difftool: \
`$ git pr show -t 18402`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18402.diff">https://git.openjdk.org/jdk/pull/18402.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18402#issuecomment-2010029479)

### CODE REVIEW

File: test/jdk/java/awt/im/bug4490692.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(() -> new TestFrame("Test Accented Chars"))
+                .build()
+                .awaitAndCheck();
+    }
+}
+
+class TestFrame extends JFrame implements KeyListener {
+    JTextField text;
```
Comment (TejeshR13) [inline test/jdk/java/awt/im/bug4490692.java:81]: text can be moved inside constructor, any reason why it is here?

File: test/jdk/java/awt/im/bug4490692.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            is generated on keypress else fails.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(40)
+                .testUI(() -> new TestFrame("Test Accented Chars"))
```
Comment (azvegint) [inline test/jdk/java/awt/im/bug4490692.java:74]: Looks like a good candidate to use `splitUIRight/Bottom` (up to you).

File: test/jdk/java/awt/im/bug4490692.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        text.addKeyListener(this);
+        Container c = getContentPane();
+        c.setLayout(new BorderLayout());
+        c.add(text, BorderLayout.CENTER);
+        setSize(300, 200);
+    }
+
+    @Override
+    public void keyPressed(KeyEvent e) {
+        if (e.getKeyCode() == 23 || e.getKeyCode() == 60) {
```
Comment (azvegint) [inline test/jdk/java/awt/im/bug4490692.java:95]: >  CASE 2: This step is automated and applicable for both
keyboards - LINUX & SOLARIS SPARC.
When Tab key is pressed it should generate a-acute (à)
character, this test automatically passes if the correct character
is generated on keypress else fails.

Unfortunately this check doesn't work for me:
`java.awt.event.KeyEvent[KEY_PRESSED,keyCode=0,keyText=Unknown keyCode: 0x0,keyChar='á',keyLocation=KEY_LOCATION_UNKNOWN,rawCode=23,primaryLevelUnicode=225,scancode=0,extendedKeyCode=0x10000e1] on javax.swing.JTextField...`

The `keycode` is 0, `rawCode` is desired `23`, but we can't get it it unless parsing the output of `e.toString()`

The `e.getKeyChar() == 0x00e1` check is still valid thought.

Comment (honkar-jdk) [reply test/jdk/java/awt/im/bug4490692.java]: @azvegint Thanks for checking it. Update the code & added the rawCode check now. Can you please re-review it.

File: test/jdk/java/awt/im/bug4490692.java (hunk 2 above)
Comment (honkar-jdk) [reply test/jdk/java/awt/im/bug4490692.java]: Agreed. Updated UI.

File: test/jdk/java/awt/im/bug4490692.java (hunk 1 above)
Comment (honkar-jdk) [reply test/jdk/java/awt/im/bug4490692.java]: Updated.

File: test/jdk/java/awt/im/bug4490692.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                .build()
+                .awaitAndCheck();
+    }
+
+    private static JComponent createUI() {
+        JPanel panel = new JPanel();
+        JTextField textField = new JTextField("", 20);
+        panel.add(new JLabel("Text field:"));
+
+        textField.addKeyListener(new KeyListener() {
```
Comment (azvegint) [inline test/jdk/java/awt/im/bug4490692.java:87]: you can use `KeyAdapter` to avoid overriding unused methods (`keyTyped`, `keyReleased`).

**Review by azvegint — APPROVED** [review]

File: test/jdk/java/awt/im/bug4490692.java (hunk 3 above)
Comment (azvegint) [reply test/jdk/java/awt/im/bug4490692.java]: Now it works for the tab case, thank you.

File: test/jdk/java/awt/im/bug4490692.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+            @Override
+            public void keyReleased(KeyEvent e) {
+            }
+        });
+
+        panel.add(textField);
+        return panel;
+    }
+}
```
[the remark this answers is not in the document]
Comment (azvegint) [reply test/jdk/java/awt/im/bug4490692.java:115]: It is a common practice:
https://stackoverflow.com/a/4700756

File: test/jdk/java/awt/im/bug4490692.java (hunk 4 above)
Comment (honkar-jdk) [reply test/jdk/java/awt/im/bug4490692.java]: Thanks! Updated.

**Review by alisenchung — APPROVED** [review]

**Pull request closed without merging**